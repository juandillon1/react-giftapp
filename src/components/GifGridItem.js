export const GifGridItem = ( {title, url} ) => {
    return (
        <div className="card animate__animated animate__bounce animate__slow">
            <img src={url} alt={title}></img>
            <p>{title}</p>
        </div>
    );
};