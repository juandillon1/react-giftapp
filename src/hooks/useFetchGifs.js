import { useState, useEffect } from "react";
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = ( categorie ) => {
    const [state, setState] = useState({
        data: [],
        loading: true,
    });
    useEffect(() => {
        getGifs( categorie )
        .then( imgs => {
            setState(
                {
                    data: imgs,
                    loading: false
                }
            );
        });
    }, [categorie])
    return state;
}