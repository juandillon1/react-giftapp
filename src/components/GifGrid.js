import { useFetchGifs } from "../hooks/useFetchGifs";
import { PropTypes } from 'prop-types';
import { GifGridItem } from './GifGridItem';
export const GifGrid = ({categories}) => {
    const {data: images, loading} = useFetchGifs(categories);

    return (
        <>
            <h3 className="animate__animated animate__bounce animate__slow"> {categories} </h3>
            {loading && <p className="animate__animated animate__flash">Cargando...</p> }
            <div className="card-grid">
            {
                images.map(img =>
                    <GifGridItem key={img.id} {...img}/>
                )
            }
            </div>
        </>
    )
};

GifGrid.propTypes = {
    categories: PropTypes.string.isRequired,
}