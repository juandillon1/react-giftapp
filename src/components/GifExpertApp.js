import {useState} from 'react'
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';
const GifExpertApp = () => {
    const [categories, setCategories] = useState(['Dragon Ball Z']);

    // const handleAdd = () => {
    //     setCategories([...categories, 'Naruto']);
    // };
    return (
        <>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={setCategories}/>
        <hr />

        <ol>
            {
                categories.map( categoria =>
                    <GifGrid categories={categoria} key={categoria}/>
                )
            }
        </ol>
        </>
    );
};

export default GifExpertApp;