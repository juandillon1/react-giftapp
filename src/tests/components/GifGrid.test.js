import { GifGrid } from "../../components/GifGrid";
import { shallow } from "enzyme";
import React from "react";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en GifGrid', () => {
    const categories = 'Goten';
    test('Debe cargar correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid categories={categories}/>);
        expect(wrapper).toMatchSnapshot();
    })
    test('Debe mostrar items cuando se cargan imagenes useFetchsGifs', () => {
        //Mocks = simular informacion
        const imgs = [{id: 'ABC', url: 'https://www.test.com/cosa.jpg', title: 'test' }]
        useFetchGifs.mockReturnValue({
            data: imgs,
            loading: false
        });
        const wrapper = shallow(<GifGrid categories={categories}/>);
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(imgs.length);
    })


})
