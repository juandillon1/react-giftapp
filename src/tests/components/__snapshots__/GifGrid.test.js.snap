// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid Debe cargar correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce animate__slow"
  >
     
    Goten
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
